### DISCUSSION TITLE
Is there a way to programmatically rehydrate the store?

### DISCUSSION THREAD

**Discussion by jclif** [ctrlplusb/easy-peasy#828] [General]
I'm building an Expo (react-native) app that has a background task that needs to update state. I have persist working so that the both the main application and the background job both pull from the same storage.

The problem I'm encountering is where the main application starts and rehydrates, then goes into a backgrounded state, whereupon a background task will set some state and persist it; unfortunately, the main app still has it's state in memory. What I'd like to do is be able to programmatically rehydrate a store every time the app goes from a backgrounded to an active state. Is there a way to do that?

**Comment by jmyrland** [comment]
As far as I know, there is currently no exposed API to rehydrate the store manually.

Rehydration is attempted whenever `createStore` is invoked - so if possible, you could attempt to recreate the store? In theory, that should work. I'm not familiar with expo / RN, so I'm not sure where to trigger the store regeneration.
